scripts/run-kani.sh

```diff
@@ -209,16 +245,44 @@ main() {
     "$kani_path" --version
 
     if [[ "$run_command" == "verify-std" ]]; then
-        echo "Running Kani verify-std command..."
-        "$kani_path" verify-std -Z unstable-options ./library \
-            -Z function-contracts \
-            -Z mem-predicates \
-            -Z loop-contracts \
-            -Z float-lib \
-            -Z c-ffi \
-            $command_args \
-            --enable-unstable \
-            --cbmc-args --object-bits 12
+        if [[ -n "$PARALLEL_INDEX" && -n "$PARALLEL_TOTAL" ]]; then
+            echo "Running as parallel worker $PARALLEL_INDEX of $PARALLEL_TOTAL"
+            get_harnesses "$kani_path"
+
+            echo "All harnesses:"
+            printf '%s\n' "${ALL_HARNESSES[@]}"
+            echo "Total number of harnesses: $HARNESS_COUNT"
+            
+            # Calculate this worker's portion
+            chunk_size=$(( (HARNESS_COUNT + PARALLEL_TOTAL - 1) / PARALLEL_TOTAL ))
+            echo "Number of harnesses this worker will run: $chunk_size"
+            
+            start_idx=$(( (PARALLEL_INDEX - 1) * chunk_size ))
+            end_idx=$(( start_idx + chunk_size ))
+            # If end_idx > HARNESS_COUNT, truncate it to $HARNESS_COUNT
+            [[ $end_idx -gt $HARNESS_COUNT ]] && end_idx=$HARNESS_COUNT
```

Comment:
This will fail if we ever were to enable `pipefail`, so I'd recommend using `if [[ ... ]]; then end_idx=$HARNESS_COUNT ; fi` instead.